holoviews/plotting/mpl/chart.py

```diff
@@ -578,59 +525,49 @@ class PointPlot(ChartPlot, ColorbarPlot):
 
     _disabled_opts = ['size']
 
-    def initialize_plot(self, ranges=None):
-        points = self.hmap.last
-        axis = self.handles['axis']
+    def init_artist(self, ax, plot_args, plot_kwargs):
+        return {'artist': ax.scatter(*plot_args, **plot_kwargs)}
 
-        ranges = self.compute_ranges(self.hmap, self.keys[-1], ranges)
-        ranges = match_spec(points, ranges)
 
-        xs = points.dimension_values(0) if len(points.data) else []
-        ys = points.dimension_values(1) if len(points.data) else []
+    def get_data(self, element, ranges, style):
+        xs, ys = (element.dimension_values(i) for i in range(2))
+        self._compute_styles(element, ranges, style)
+        return (xs, ys), style, {}
 
-        style = self.style[self.cyclic_index]
-        cdim = points.get_dimension(self.color_index)
+
+    def _compute_styles(self, element, ranges, style):
+        cdim = element.get_dimension(self.color_index)
         color = style.pop('color', None)
         if cdim:
-            cs = points.dimension_values(self.color_index)
+            cs = element.dimension_values(self.color_index)
             style['c'] = cs
             if 'clim' not in style:
                 clims = ranges[cdim.name]
                 style.update(vmin=clims[0], vmax=clims[1])
         else:
             style['c'] = color
-        edgecolor = style.pop('edgecolors', style.pop('edgecolor', 'none'))
-
-        if points.get_dimension(self.size_index):
-            style['s'] = self._compute_size(points, style)
-
-        legend = points.label if self.show_legend else ''
-        scatterplot = axis.scatter(xs, ys, zorder=self.zorder, label=legend,
-                                   edgecolors=edgecolor, **style)
-        self.handles['artist'] = scatterplot
+        style['edgecolors'] = style.pop('edgecolors', style.pop('edgecolor', 'none'))
 
-        return self._finalize_axis(self.keys[-1], ranges=ranges)
+        if element.get_dimension(self.size_index):
+            sizes = element.dimension_values(self.size_index)
+            ms = style.pop('s') if 's' in style else plt.rcParams['lines.markersize']
```

Comment:
Happy to use ``rcParams`` here? I thought ``rcParams`` could sometimes be tricky...